backend/tests/test_reservation_check_contracts.py

```
@@ -0,0 +1,148 @@
+
+from database import Base, SessionLocal, engine
+from models import Client, ClientReservations, Reservation, Room, Structure
+from routes import reservation_routes
+from routes.reservation_routes import UPLOAD_TOKEN_SALT, reservation_bp
+
+
+class _FailingSession:
+    """Session stub that raises SQLAlchemyError on query usage."""
+
+    def query(self, *_args, **_kwargs):
+        raise SQLAlchemyError("db down")
```

Ruff (0.15.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)